app/models/template_input.rb

```diff
@@ -91,11 +91,12 @@ def resolved_value
 
   class UserInputResolver < InputResolver
     def ready?
-      !input_value.nil?
+      @renderer.invocation
     end
 
     def resolved_value
-      input_value.value
+      raise(ValueNotReady, _("Required input not specified")) if @input.required? && input_value.try(:value).blank?
```

Comment:
well, raise can remain in here as well but resolved_value should never be called if ready is false (I guess this is defensive check)

by accept blank value I mean that `input_value.try(:value).blank?` is also true for empty string which can be valid value I think, so `input_value.nil?` should be used instead?